tests/test_taskgroups.py

```diff
@@ -486,6 +486,37 @@ async def test_cancel_from_shielded_scope():
             await sleep(0.01)
 
 
+@pytest.mark.parametrize('anyio_backend', ['asyncio'])
+async def test_cancel_host_asyncgen():
+    done = False
+
+    async def host_task():
+        nonlocal done
+        async with create_task_group() as tg:
+            with open_cancel_scope(shield=True) as inner_scope:
+                assert inner_scope.shield
+                tg.cancel_scope.cancel()
+
+            with pytest.raises(get_cancelled_exc_class()):
+                await sleep(0)
+
+            with pytest.raises(get_cancelled_exc_class()):
+                await sleep(0)
+            done = True
+
+    async def host_agen_fn():
+        await host_task()
+        yield
+        pytest.fail("host_agen_fn should only be __anext__ed once")
+
+    host_agen = host_agen_fn()
```

Comment:
For consistency with other tests.
```suggestion
    done = False
    host_agen = host_agen_fn()
```